spyder/widgets/ipythonconsole/shell.py

```diff
@@ -283,7 +284,7 @@ def _banner_default(self):
 
     def _kernel_restarted_message(self, died=True):
         msg = _("Kernel died, restarting") if died else _("Kernel restarting")
-        self.sig_kernel_restarted.emit(msg)
```

Comment:
Please revert this change. This is not related to the things you're trying to solve in this PR.